src/Renderer.cpp

```diff
@@ -0,0 +1,169 @@
+#include "Renderer.h"
+
+#include "AssetManager.h"
+
+#include <glm/gtc/matrix_transform.hpp>
+
+Renderer::Renderer(AssetManager& n_asset_manager)
+    : time(0.0f),
+      asset_manager(n_asset_manager) {
+    EventSystem::add_event_handler(EventType::LOAD_EVENT, &Renderer::load, this);
+    EventSystem::add_event_handler(EventType::KEYPRESS_EVENT, &Renderer::handle_key_press, this);
+}
+
+void Renderer::update() {
+    example_objects[1].apply_transform(glm::rotate(glm::mat4x4(), 0.01f, glm::vec3(1, 1, 1)));
+
+    //time += 0.01f;
+
+    //This should be moved to a different system
+    SDL_Event event;
+
+    while (SDL_PollEvent(&event)) {
+        switch (event.type) {
+            case SDL_QUIT:
+                std::cout << "SDL_QUIT was called" << std::endl;
+                SDL_Quit();
+                break;
+        }
+    }
+}
+
+void Renderer::load(const Event& e) {
+    init_window();
+
+    example_shader.load_shader("SimpleVertexShader.vertexshader", "SimpleFragmentShader.fragmentshader");
+
+    auto mesh = asset_manager.get_mesh_asset("Ship.obj");
+
+    example_objects.emplace_back();
+    example_objects[0].set_mesh(mesh);
+    example_objects[0].apply_transform(glm::translate(glm::mat4x4(), glm::vec3(0, -2, 0)));
+
+    example_objects.emplace_back();
+    example_objects[1].set_mesh(mesh);
+    example_objects[1].apply_transform(glm::translate(glm::mat4x4(), glm::vec3(1, 2, 1)));
+
+    setup_cameras();
+}
+
+void Renderer::handle_key_press(const Event& e) {
+    int player_id = e.get_value<int>("player_id", -1);
+    int key = e.get_value<int>("key", -1);
+    int value = e.get_value<int>("value", 0);
+
+    glm::mat4 transform;
+
+    switch (key) {
+        case SDLK_LEFT:
```

Comment:
Just a warning: same keys are mapped for  audio testing. Might be a little annoying when we're testing the camera